src/modules/cxxpiper.cpp

```diff
@@ -0,0 +1,1223 @@
+    MOD_OPTION_1_STR(ModelPath)
+    MOD_OPTION_1_STR(ConfigPath)
+    MOD_OPTION_1_STR(SoundIconFolder)
+    MOD_OPTION_1_INT(SoundIconVolume)
+    // SIC: see below regarding NONE and ALL
+    MOD_OPTION_1_STR(PunctSome)
+    MOD_OPTION_1_STR(PunctMost)
+    MOD_OPTION_1_STR(ESpeakNGDataDirPath)
+
+    int module_load(void)
+    {
+	INIT_SETTINGS_TABLES();
+	REGISTER_DEBUG();
+	MOD_OPTION_1_INT_REG(UseCUDA, 0);
+	MOD_OPTION_1_STR_REG(ModelPath, "");
+	MOD_OPTION_1_STR_REG(ConfigPath, "");
+	MOD_OPTION_1_STR_REG(SoundIconFolder, "/usr/share/sounds/sound-icons/");
+	MOD_OPTION_1_INT_REG(SoundIconVolume, 0);
+	// NB:  NONE and ALL are constant so no config.  NONE means omit as much as possible
+	// while ALL  means omit as little as possible.
+	// NB: Don't provide defaults, since empty string means fallback to SD's definitions.
+	MOD_OPTION_1_STR_REG(PunctSome, "");
+	MOD_OPTION_1_STR_REG(PunctMost, "");
+	MOD_OPTION_1_STR_REG(ESpeakNGDataDirPath, "/usr/share/espeak-ng-data/");
+	module_register_available_voices();
+	module_register_settings_voices();
+	return 0;
+    }
+
+    int module_init(char **status_info)
+    {
+	module_audio_set_server();
+	module_audio_init(status_info);
+	runConfig.useCuda = UseCUDA;
+	runConfig.modelPath = filesystem::path(ModelPath);
+	runConfig.modelConfigPath = filesystem::path(ConfigPath);
+	runConfig.eSpeakDataPath = filesystem::path(ESpeakNGDataDirPath);
+	runConfig.outputPath = nullopt;
+	char *default_voice = module_getdefaultvoice();
+	runConfig.defaultVoiceName = default_voice;
+	runConfig.speakerId = cxxpiper_voice_name_to_speaker_id(runConfig.defaultVoiceName);
+	DBG("Default Voice is %s", runConfig.defaultVoiceName);
+	cxxpiper::loadVoice(piperConfig, runConfig.modelPath.string(), runConfig.modelConfigPath.string(),
+				voice, runConfig.speakerId, runConfig.useCuda);
+	    if (voice.phonemizeConfig.phonemeType == piper::eSpeakPhonemes) {
+		piperConfig.eSpeakDataPath = runConfig.eSpeakDataPath.value().string();
+	    }
+	    else {
+		piperConfig.useESpeak = false;
+	    }
+	    cxxpiper::initialize(piperConfig);
+	    cxxpiper_voice_list = cxxpiper_allocate_voice_list();
+	    cxxpiper_alloc_cbuf();
+	    *status_info = g_strdup(DBG_MODNAME " Initialized successfully.");
+	    return 0;
+	    }
+
+	    SPDVoice **module_list_voices(void)
+    {
+	return cxxpiper_voice_list;
+    }
+
+    void module_speak_sync(const char *data, size_t bytes, SPDMessageType msgtype)
+    {
+	stop_requested = 0;
+	UPDATE_STRING_PARAMETER(voice.language, cxxpiper_set_language);
+	UPDATE_PARAMETER(voice_type, cxxpiper_set_voice_type);
+	UPDATE_STRING_PARAMETER(voice.name, cxxpiper_set_synthesis_voice);
+	module_speak_ok();
+	switch (msgtype) {
+	case SPD_MSGTYPE_CHAR:
+	case SPD_MSGTYPE_KEY:
+	case SPD_MSGTYPE_SPELL:
+	    break;
+	case SPD_MSGTYPE_TEXT: {
+	    cxxpiper_handle_text(data);
+	    break;
+	}
+	case SPD_MSGTYPE_SOUND_ICON: {
+	    cxxpiper_handle_sound_icon(data);
+	    break;
+	}
+	}
```

Comment:
This double `}` is odd, better either indent the whole content of the `switch`, or the content of each `case`.